### PR TITLE
Switch this repo from OWNERS to CODEOWNERS

### PR DESCRIPTION
**Pull request by geeknoid**
Align this repo with the rest of the Istio repos.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: CODEOWNERS
Comment (cjwagner) [inline CODEOWNERS]: Is this going to start requesting my review on every ProwJob config change?
I think oncall should have permission to approve changes to these files when needed, but most job config changes are outside the scope of oncall and are the responsibility of the job authors so oncall shouldn't be requested for review.

Comment (fejta) [reply CODEOWNERS]: Yes I do not want to get assigned as a reviewer on every PR in this repo. For the most part I never want to see anything in prow/cluster/jobs, want this delegated to the subject matter experts

Comment (howardjohn) [reply CODEOWNERS]: This is a problem we have on istio/istio as well. We want some people to be able to approve anything, but not actually be requested for review on every PR. I don't think we have a way to allow oncall type people to approve job changes without making them marked for review on every pr

Comment (fejta) [reply CODEOWNERS]: Use the better designed for large repos OWNERS files? :smiling_imp: 

In this case I'd be happy to just not be an approver for anything under prow/cluster/jobs and let other people do these reviews.

Also I suspect if we make Cole and I (etc) maintainers of this repo we are automatically approvers for everything

**Comment by geeknoid** [comment]
My original CODEOWNERS  was a direct translation of the OWNERS files, giving the same ownership coverage to the same individuals. I've updated this now to remove engprod folks from the job directories.

PTAL.

File: CODEOWNERS
Comment (PiotrSikora) [inline CODEOWNERS:12]: Could you add me to `/prow/cluster/jobs/istio/proxy/`? Thanks!

Comment (geeknoid) [reply CODEOWNERS:12]: Done.

**Review by fejta — APPROVED** [review]
/restest

File: prow/config.yaml
Comment (fejta) [inline prow/config.yaml:200]: This is not correct, test-infra is not part of the istio release. Please leave `protect: true` on as before

**Review by fejta — APPROVED** [review]

File: prow/plugins.yaml
Comment (howardjohn) [inline prow/plugins.yaml:80]: don't need lgtm either

**Comment by fejta** [comment]
This broke prow, please revert

**Pull request merged**
